addon/index.js

```diff
@@ -63,45 +91,35 @@ function setupApp() {
     app.send('addon-self:upgraded');
   }
 
-  app.on('update-check', function() {
-    app.send('addon-updates', state.updates);
+  app.on('install-experiment', function(msg) {
+    installExperiment(msg);
   });
 
-  app.on('update-approve', function(msg) {
-    const msgIds = msg.map(m => m.id);
-    triggerInstalls(updates.filter(u => require('array-includes')(msgIds, u.id)));
+  app.on('uninstall-experiment', function(msg) {
+    uninstall(msg);
   });
 
   app.on('uninstall-all', function() {
-    uninstall(store.experiments);
-  });
-
-  app.on('uninstall', function(msg) {
-    uninstall(msg);
+    Object.keys(store.experiments).forEach(key => uninstall({addon_id: key}));
   });
 
   app.on('loaded', function() {
-    app.send('addon-available', {
-      experiments: store.experiments,
-      updates: state.updates
+    getInstalledAddons(addons => {
+      app.send('addon-available', {
+        installed: addons
+      });
     });
   });
 }
 
-function uninstall(toUninstall) {
-  toUninstall.forEach(u => {
-    AddonManager.getAddonByID(u.id, function(a) {
-      a.uninstall();
-    });
-  });
+function uninstall(experiment) {
+  AddonManager.getAddonByID(experiment.addon_id, a => a.uninstall());
 }
 
-function triggerInstalls(approvedUpdates) {
-  approvedUpdates.forEach(u => {
-    AddonManager.getInstallForURL(u.url, function(install) {
-      install.install();
-    }, 'application/x-xpinstall');
-  });
+function installExperiment(experiment) {
+  AddonManager.getInstallForURL(experiment.xpi_url, install => {
```

Comment:
Ditto here: this could be used to install any addon. let's make sure it's an ideatown addon